ARCHITECTURE.md

```diff
@@ -133,37 +136,38 @@ serialisations for PATCH requests.
 
 ## `src/access/*`, `src/acl/*` and `src/acp/*`
 
-At first, there was Web Access Control. And although people often referred to it
-as Access Control Lists, He saw it was Good.
+At first, there was one [Solid authorization mechanism](https://solidproject.org/TR/protocol#authorization), 
+Web Access Control ([WAC](https://solidproject.org/TR/wac)), and people often 
+referred to it as Access Control Lists.
 
-Or at least, it was OK. It was a pretty flexible system that allows for quite a
-few different access use cases, although that meant simple use cases like "allow
-this person to view this Resource" were relatively cumbersome to define: after
-all, you'd have to take into account all kinds of access that could be defined
-earlier. But that's OK: `src/acl/*` contains code that does all that nasty
-stuff.
+WAC is an access control system to set authorization conditions on HTTP resources 
+using the Access Control List (ACL) model. It allows for quite a few different use 
+cases, although simple use cases like "allow this person to view this 
+Resource" were relatively cumbersome to define given the reliance on inheritance
+to effective establish the access control for a given agent: you'd have to take into 
+account all kinds of access that could be defined earlier. But that's OK: `src/acl/*` 
+contains code that does all that nasty stuff.
 
-But then Inrupt ate from the tree of knowledge, and proposed a new mechanism:
-Access Control Policies. Which, in the end, is also just a bunch of data written
-to a particular bunch of Resources. `src/acp/*` contains APIs that are
-essentially wrappers around our existing APIs for reading and writing data, but
-using ACP terminology.
+In order to solve these (and other) intrinsic limitations of WAC, Inrupt proposed 
+a new mechanism: Access Control Policies ([ACP](https://github.com/solid/authorization-panel/blob/main/proposals/acp/index.md)). 
+ACP is more flexible and stores authorization information across a particular set 
+of Resources. `src/acp/*` contains APIs that are essentially wrappers around our 
+existing APIs for reading and writing data, but using ACP terminology.
 
-However, the ACP proposal is even more flexible then WAC, making simple use
-cases even more cumbersome to define. And to make matters worse, it had to
-co-exist next to WAC, which doesn't seem to be going anywhere anytime soon.
+However, the ACP proposal has to co-exist next to WAC for the time being, which does
```

Comment:
This changes the messaging, which is intended to contributors to the codebase, and which I think is relevant, that inherently, ACPs flexibility makes low-level use cases complex.